components/api/src/test/java/com/hotels/styx/api/extension/service/spi/MockContext.java

```diff
@@ -19,9 +19,12 @@
 
 import java.net.InetSocketAddress;
 import java.util.Optional;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
 
 class MockContext implements HttpInterceptor.Context {
     static final HttpInterceptor.Context MOCK_CONTEXT = new MockContext();
+    final Executor executor = Executors.newSingleThreadExecutor();
```

Comment:
Should we replace this with Runnable::run as discussed?

Reply:
👍